packages/sanity/src/core/form/inputs/PortableText/PortableTextInput.tsx

```diff
@@ -145,26 +160,35 @@ export function PortableTextInput(props: PortableTextInputProps): ReactNode {
   // It's a bit ugly right here, but it's a rather simple way to support the Presentation tool without
   // having to change the PTE's internals.
   const setFocusPathFromEditorSelection = useCallback(
-    (focusPath: Path) => {
+    (selection: EditorSelection) => {
+      const focusPath = selection?.focus.path
+
+      if (!focusPath) return
+
       // Test if the focusPath is pointing directly to a span
       const isSpanPath =
         focusPath.length === 3 && // A span path is always 3 segments long
         focusPath[1] === 'children' && // Is a child of a block
         isKeySegment(focusPath[2]) && // Contains the key of the child
         !portableTextMemberItems.some(
           (item) => isKeySegment(focusPath[2]) && item.member.key === focusPath[2]._key,
-        ) // Not an inline object (it would be a member in this list, where spans are not). By doing this check we avoid depending on the value.
-      if (isSpanPath) {
-        // Append `.text` to the focusPath
-        onPathFocus(focusPath.concat('text'))
-      } else {
-        // Call normally
-        onPathFocus(focusPath)
-      }
+        )
+
```

Comment:
Can we please have this with explicit comments as it were? It's explaining the "hack".